src/config/config.c

```diff
@@ -106,12 +107,16 @@ static const char* prev_item(const char* opt, const char* end)
     {
         beg--;
     }
-    int len = (int)(end - beg);
+    assert(beg <= end);
+    size_t len = (size_t)(end - beg);
     if(len > SECTION_NAME_MAX)
     {
         len = SECTION_NAME_MAX;
     }
-    strncpy(opt_item, beg, len);
+    if(len > 0)
```

Comment:
This guard is not needed since it is equivalent to `len != 0`, and if `len == 0`, `strncpy()` will do nothing anyway.